service/src/test/java/bio/terra/cbas/dependencies/wds/TestScheduledWdsClientRefresher.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.cbas.dependencies.wds;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.ws.rs.client.Client;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TestScheduledWdsClientRefresher {
+
+  @Test
+  void canGetCurrentClient() {
+    ScheduledWdsClientRefresher scheduledWdsClientRefresher = new ScheduledWdsClientRefresher();
+    assert (scheduledWdsClientRefresher.getCurrentClient() != null);
+  }
+
+  @Test
+  void buildsUpRetiredClients() {
+    ScheduledWdsClientRefresher scheduledWdsClientRefresher = new ScheduledWdsClientRefresher();
+
+    List<Integer> clientHashes = new ArrayList<>();
+
+    for (int i = 0; i < 10; i++) {
+      clientHashes.add(scheduledWdsClientRefresher.getCurrentClient().hashCode());
+      scheduledWdsClientRefresher.refreshClient();
+    }
```

Comment:
I think `generate()` could work here too since `i` is unneeded
```
    List<Integer> clientHashes = Stream.generate(() -> {
      int hash = scheduledWdsClientRefresher.getCurrentClient().hashCode()
      scheduledWdsClientRefresher.refreshClient();
      return hash;
    }).limit(10);
```